vnext/Microsoft.ReactNative/Views/ConfigureBundlerDlg.cpp

```diff
@@ -0,0 +1,180 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "pch.h"
+#include "ConfigureBundlerDlg.h"
+#include "DevServerHelper.h"
+#include "IReactDispatcher.h"
+#include "UI.Xaml.Controls.Primitives.h"
+#include "UI.Xaml.Controls.h"
+#include "UI.Xaml.Input.h"
+#include "UI.Xaml.Markup.h"
+#include "Unicode.h"
+#include "Utils/Helpers.h"
+#include "winrt/Windows.UI.Xaml.Interop.h"
+
+namespace Microsoft::ReactNative {
+
+ConfigureBundlerDlg::ConfigureBundlerDlg(Mso::WeakPtr<Mso::React::IReactHost> weakReactHost)
+    : m_weakReactHost(std::move(weakReactHost)) {
+  if (auto reactHost = m_weakReactHost.GetStrongPtr()) {
+    auto uiDispatcher = winrt::Microsoft::ReactNative::implementation::ReactDispatcher::GetUIDispatcher(
+        reactHost->Options().Properties);
+    uiDispatcher.Post([pThis = get_strong()]() { pThis->CreateAndShowUI(); });
+  }
+}
+
+void ConfigureBundlerDlg::CreateAndShowUI() noexcept {
+  if (auto reactHost = m_weakReactHost.GetStrongPtr()) {
+    const winrt::hstring xamlString =
+        LR"(
+  <StackPanel
+    xmlns='http://schemas.microsoft.com/winfx/2006/xaml/presentation'
+    xmlns:x='http://schemas.microsoft.com/winfx/2006/xaml'
+    Background='{ThemeResource ContentDialogBackground}'
+    HorizontalAlignment='Center'
+    MinWidth="{ThemeResource ContentDialogMinWidth}"
+    MaxWidth="{ThemeResource ContentDialogMaxWidth}"
+    MinHeight="{ThemeResource ContentDialogMinHeight}"
+    MaxHeight="{ThemeResource ContentDialogMaxHeight}"
+    Margin='-10'
+    Padding="{ThemeResource ContentDialogPadding}"
+  >
+    <StackPanel.Transitions>
+      <TransitionCollection>
+        <EntranceThemeTransition />
+      </TransitionCollection>
+    </StackPanel.Transitions>
+    <TextBlock FontSize="20" Margin="{ThemeResource ContentDialogTitleMargin}">Configure Bundler</TextBlock>
+    <TextBox x:Name="Host" Header="Bundler host:" PlaceholderText="localhost"/>
+    <TextBox x:Name="Port" Header="Bundler port:" PlaceholderText="8081" Margin="0,4,0,0"/>
+    <TextBox x:Name="EntryPoint" Header="entrypoint:" PlaceholderText="index.windows" Margin="0,4,0,0"/>
+    <StackPanel Orientation="Horizontal" HorizontalAlignment="Right" Margin="{ThemeResource ContentDialogCommandSpaceMargin}">
+      <Button x:Name="Apply" Style="{StaticResource AccentButtonStyle}" Content="Apply Changes" />
```

Comment:
>Apply [](start = 22, length = 5)

Should we put a warning that it will reload the app, loosing the current app state?